add ai readme (#1951)
microsoft/azure_arc: docs/azure_jumpstart_ag/contoso_supermarket/ai/_index.md

> To this end, Contoso has developed an AI mechanism that empowers them to enhance the customer experience in their stores. The objective is to accurately detect the number of people waiting at different checkout lanes and enable proactive actions based on this data.

Reviewer: ```suggestion
To this end, Contoso Supermarket has developed an artificial intelligence (AI) mechanism that empowers them to enhance the customer experience in their stores. By implementing such a mechanism, their objective is to accurately detect the number of people waiting at different checkout lanes and enable proactive actions based on this data.
```

> ## Architecture

Reviewer: I am missing some context on the AI portion. What technology is being used and why.

   ![Screenshot showing the Contoso Supermarket's queue monitoring service](./img/ai_diagram.png)

Contoso Supermarket's DevOps team has adopted GitOps methodologies which allows them to use Git as the single source of truth for managing infrastructure and application deployments. It involves declarative definitions of infrastructure and application configurations, which are stored in Git repositories. The GitOps pipeline automatically detects any changes made to the repositories and triggers the necessary actions to deploy the changes to the target environments.

One of the main benefits of GitOps is its ability to provide a consistent and auditable deployment process. All changes to the infrastructure and application configurations are tracked and versioned in Git, making it easy to roll back to a previous state if necessary. The declarative nature of GitOps also enables the automation of the deployment process, which reduces the risk of human error and streamlines the entire process.

### GitOps workflow

> Contoso has deployed GitOps configurations for all of their services to make it easier for their developers to focus on code and streamline all the code build, test, and deployment activities. Flux v2 is the main driver for Contoso's Supermarket GitOps workflow by monitoring the Git repository holding all the clusters' configurations and applications code for changes and automatically updates the deployed resources accordingly. This makes it easy for teams to manage their infrastructure as code, using Git as the single source of truth.

Reviewer: ```suggestion
Contoso has deployed GitOps configurations for all of their services to make it easier for their developers to focus on code and streamline all the code build, test, and deployment activities. Flux v2 is the main driver for Contoso's GitOps workflow by monitoring the Git repository holding all the clusters' configurations and applications code for changes and automatically updating the deployed resources accordingly. This makes it easy for teams to manage their infrastructure as code, using Git as the single source of truth.
```

Reviewer: Add some more background on Flux v2 and the relations to Azure Arc.  Also, incorporate a learn more link. This will be a great one to use:
https://learn.microsoft.com/azure/cloud-adoption-framework/scenarios/hybrid/arc-enabled-kubernetes/eslz-arc-kubernetes-cicd-gitops-disciplines

    ![Screenshot showing the developer folder structure](./img/repo_developer_structure.png)

- Within the _operations_ folder there is also a folder for each application's GitOps configuration. This is where Contoso Supermarket's DevOps team manages how the applications are deployed to the different environments.

    ![Screenshot showing the operations folder structure](./img/repo_operations_structure.png)

- Each application has a folder for [Helm](https://helm.sh/docs/) _charts_ where the Kubernetes manifests for each application are located and a folder for _releases_ where the Helm Releases for each application and each environment is located. This way the DevOps team can control the promotion of each version of the applications across the GitOps workflow on different environments and also enable/disable features created by the developers as needed.

    ![Screenshot showing the helm folder structure](./img/repo_operations_helm_structure.png)

    ![Screenshot showing the helm releases folder structure](./img/repo_operations_helmreleases_structure.png)

## Store Manager Experience

> Contoso leverages their existing CCTV cameras by seamlessly integrating them into the store manager interface. This integration enables Store Managers to take immediate action, such as opening or closing lanes, based on the real-time data provided by the cameras. All data processing occurs locally using an edge device, ensuring compliance with corporate policies and safeguarding data privacy.

Reviewer: ```suggestion
Contoso leverages their existing Closed-circuit television (CCTV) cameras by seamlessly integrating them into the store manager interface. This integration enables Store Managers to take immediate action, such as opening or closing lanes, based on the real-time data provided by the cameras. All data processing occurs locally using an edge device, ensuring compliance with corporate policies and safeguarding data privacy.
```

> ![POS Bookmar](./img/control-center-menu.png)

Reviewer: ```suggestion
![POS Bookmark](./img/control-center-menu.png)
```

![Screenshot of disabled live view button](./img/disabled-live-view.png)

## DevOps team experience

> The Contoso Supermarket's DevOps team has received a request from the _Chicago_ store managers that they need the live view feature enabled in their store as their queues are getting longer throughout peak hours through the day.

Reviewer: ```suggestion
Contoso's DevOps team has received a request from the _Chicago_ store managers that they need the "Live View" feature enabled in their store as their queues are getting longer throughout peak hours throughout the day.
```

- Connect to the Client VM `Ag-VM-Client` using the instructions in the [Deployment Guide](https://github.com/microsoft/azure_arc/blob/jumpstart_ag/docs/azure_jumpstart_ag/contoso_supermarket/deployment/_index.md#connecting-to-the-agora-client-virtual-machine).

- Open VSCode from the desktop shortcut.

    ![Screenshot showing the the VSCode icon on the desktop](./img/open_vscode.png)

- Bring up the VSCode command palette.

    ![Screenshot showing the opening the command palette in VSCode](./img/vscode_command_palette.png)

- Select the option to open a folder in a dev container from the command palette.

    ![Screenshot showing opening a folder in a dev container](./img/vscode_dev_container.png)

- Browse to the cloned repository on the Client VM located at _C:\Ag\AppsRepo\jumpstart-agora-apps_.

    ![Screenshot showing the cloned repository on the client VM](./img/vscode_dev_container_open_folder.png)

- Select the _Ubuntu_ operating system for your dev container.

>     ![Screenshot showing the operating system for the dev container](./img/vscode_dev_container_os.png)
> 
>     ![Screenshot showing the operating system creation for the dev container](./img/vscode_dev_container_os_create.png)
> 
> - No need for any additional features to install, so click _Ok_.
> 
>     ![Screenshot showing the additional operating system features to install](./img/vscode_dev_container_os_options.png)
> 
> - Click on _Trust folder and continue_, now you can see the cloned repository opened in VSCode, in the _Ubuntu_ dev container.
> 
>     ![Screenshot showing the trust folder prompt in VSCode](./img/vscode_dev_container_pos_app.png)
> 
>     ![Screenshot showing the cloned repository opened in the dev container](./img/vscode_dev_container_trust_folder.png)
> 
> - Click on the GitHub icon in the VSCode toolbar, click on _Manage Unsafe Repositories_ and select the _jumpstart-agora-apps_ repository to whitelist it.
> 
>     ![Screenshot showing the trust repository prompt in VSCode](./img/vscode_dev_container_trust_repository.png)
> 
> - Switch to the _canary_ branch to enable the "Live View"feature on the _Chicago_ Kubernetes cluster
> 
>     ![Screenshot showing switching to the canary branch](./img/vscode_canary_branch.png)
> 
> - Navigate to the file _contoso_supermarket/operations/contoso_supermarket/releases/queue-monitoring-frontend/canary/chicago.yaml_. You can see that "Live View" is disabled
> 
>     ![Screenshot showing the navbar.html file](./img/vscode_canary_live_view_disabled.png)
> 
> - Change the value to _True_ to enable the "Live View" feature

Reviewer: ```suggestion
    ![Screenshot showing the operating system for the dev container](./img/vscode_dev_container_os.png)

    ![Screenshot showing the operating system creation for the dev container](./img/vscode_dev_container_os_create.png)

- No need for any additional features to install, so click _Ok_.

    ![Screenshot showing the additional operating system features to install](./img/vscode_dev_container_os_options.png)

- Click on _Trust folder and continue_, now you can see the cloned repository opened in VSCode, in the _Ubuntu_ dev container.

    ![Screenshot showing the trust folder prompt in VSCode](./img/vscode_dev_container_pos_app.png)

    ![Screenshot showing the cloned repository opened in the dev container](./img/vscode_dev_container_trust_folder.png)

- Click on the GitHub icon in the VSCode toolbar, click on _Manage Unsafe Repositories_, and select the _jumpstart-agora-apps_ repository to whitelist it.

    ![Screenshot showing the trust repository prompt in VSCode](./img/vscode_dev_container_trust_repository.png)

- Switch to the _canary_ branch to enable the "Live View" feature on the _Chicago_ Kubernetes cluster

    ![Screenshot showing switching to the canary branch](./img/vscode_canary_branch.png)

- Navigate to the file _contoso_supermarket/operations/contoso_supermarket/releases/queue-monitoring-frontend/canary/chicago.yaml_. You can see that "Live View" is disabled

    ![Screenshot showing the navbar.html file](./img/vscode_canary_live_view_disabled.png)

- Change the value to _True_ to enable the "Live View" feature
```

    ![Screenshot showing the navbar.html file](./img/vscode_canary_live_view_enabled.png)

> - After a couple of seconds, Flux v2 should detect this new change and you should start seeing pod recreation activity on the _Chicago_ Kubernetes clusters

Reviewer: ```suggestion
- After a couple of seconds, the Flux operator should detect the change and you should start seeing pod recreation activity on the _Chicago_ Kubernetes cluster.
```

    ![Screenshot showing pods terminating in the canary cluster](./img/live_view_containers.png)

> - After refreshing the store manager view, we can see that the "Live View" feature is enabled and the "Live View" button is not greyed out anymore.

Reviewer: ```suggestion
- Upon refreshing the store manager view, you can see that the "Live View" feature is enabled and the "Live View" button is not greyed out anymore.
```
